### PR TITLE
Kops Secrets on Nodes

### PR DESCRIPTION
**Pull request by gambol99**
The current implementation permits nodes access to /secrets/* though the nodes themselve do [not](https://github.com/gambol99/kops/blob/secrets/nodeup/pkg/model/secrets.go#L77-L79) require access. This PR changed the ACL on the iam policy to deny access for nodes to /secrets/*

- removed the /secrets from node iam policy
- ordered the list alphabetically 

one @question ... when would [this](https://github.com/gambol99/kops/blob/secrets/nodeup/pkg/model/secrets.go#L61-L74) be used? ... and given you can managed dockerconfig in k8s is this needed ...

### CODE REVIEW

**Comment by gambol99** [comment]
@KashifSaadat @justinsb @chrislovecnm

File: pkg/model/iam/iam_builder_test.go [hunk 1]
```diff
@@ -141,14 +141,14 @@ func TestS3PolicyGeneration(t *testing.T) {
...
 						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/instancegroup/*",
 						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/pki/issued/*",
-						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/pki/ssh/*",
 						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/pki/private/kube-proxy/*",
 						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/pki/private/kubelet/*",
-						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/secrets/*",
-						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/cluster.spec",
-						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/config",
+						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/pki/ssh/*",
+						"arn:aws:s3:::bucket-name/cluster-name.k8s.local/secrets/dockerconfig",
```
Comment (KashifSaadat) [inline pkg/model/iam/iam_builder_test.go]: Was the `dockerconfig` secret pushed up in a certain scenario? I couldn't find this secret for my clusters.

**Comment by KashifSaadat** [comment]
1 minor comment, otherwise LGTM.

/assign justinsb

File: pkg/model/iam/iam_builder_test.go (hunk 1 above)
Comment (gambol99) [reply pkg/model/iam/iam_builder_test.go]: I stumbled on [this](https://github.com/kubernetes/kops/blob/master/nodeup/pkg/model/secrets.go#L61-L74) and figured it was an option somewhere, though admittedly I didn't check if it was legacy or actually being used .. so i'll dig about

**Comment by justinsb** [comment]
So the dockerconfig secret is used if you want to give a blanket permission to a docker registry.  The use cases that I know of:

* If you want to use hooks from a protected registry
* If you want to run k8s images from a protected registry
* If you just don't want to grant each pod its own docker secret

Personally, I just grant each pod its own docker secret, but it is attractive to just easily grant access for the whole cluster to a private registry.

File: pkg/model/iam/iam_builder_test.go (hunk 1 above)
Comment (justinsb) [reply pkg/model/iam/iam_builder_test.go]: It's actually fairly new I believe.  There's an explicit command to create it (kops create secret dockerconfig).  I don't know that I would actively recommend it, but there is nothing wrong with doing so.

**Pull request merged**